pyccel/codegen/printing/fcode.py

```diff
@@ -1089,23 +1256,40 @@ def _print_Assign(self, expr):
         if isinstance(rhs, (PythonRange, Product)):
             return ''
 
-        if isinstance(rhs, (NumpyArray, NumpyLinspace, NumpyDiag, NumpyCross,\
-						NumpyWhere, PyccelArraySize)):
+        if isinstance(rhs, (PythonLen, NumpyRandint)):
+            rhs_code = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code, rhs_code)
+
+        if isinstance(rhs, PythonInt):
+            rhs = expr.rhs.fprint(self._print)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
+
+        if isinstance(rhs, (NumpyWhere, NumpyCross, NumpyComplex, NumpyReal, NumpyArray, NumpyLinspace)):
+            rhs = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
+
+        if isinstance(rhs, NumpyDiag):
+            return self._print(rhs, expr.lhs)
+
+        if isinstance(rhs, PyccelArraySize):
             return rhs.fprint(self._print, expr.lhs) + '\n'
 
-        if isinstance(rhs, (NumpyFull, NumpyFullLike, NumpyEmptyLike,\
-						NumpyZerosLike, NumpyOnesLike, NumpyRand)):
+        if isinstance(rhs, NumpyRand):
+            return 'call random_number({0})\n'.format(self._print(expr.lhs))
 
-            stack_array = False
-            if self._current_function:
-                name = self._current_function
-                func = self.get_function(name)
-                lhs_name = expr.lhs.name
-                vars_dict = {i.name: i for i in func.local_vars}
-                if lhs_name in vars_dict:
-                    stack_array = vars_dict[lhs_name].is_stack_array
+        if isinstance(rhs, NumpyFull):
+            stmts = []
+            if rhs.fill_value is not None:
+                code_init = '{0} = {1}'.format(lhs_code, self._print(rhs))
+                stmts.append(code_init)
+            if len(stmts) == 0:
+                return ''
+            else:
+                return '\n'.join(stmts) + '\n'
 
-            return rhs.fprint(self._print, expr.lhs, stack_array)
+        if isinstance(rhs, (NumpyFullLike, NumpyEmptyLike,\
+						NumpyZerosLike, NumpyOnesLike)):
+            return rhs.fprint(self._print, expr.lhs)
```

Comment:
Why does this `fprint` still exist?

Reply:
because their fprints don't exit in ast.numpyext, Is the printer of NumpyFull will be enough for those? if so they will be handled with _print_NumpyFull with some modifications.